src/trs/rewrite.rs

```diff
@@ -309,6 +309,87 @@ impl TRS {
             Ok(trs)
         }
     }
+/// Move a Rule from one place in the TRS to another at random, excluding the background.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #[macro_use] extern crate polytype;
+    /// # extern crate programinduction;
+    /// # extern crate rand;
+    /// # extern crate term_rewriting;
+    /// # use programinduction::trs::{TRS, Lexicon};
+    /// # use rand::{thread_rng};
+    /// # use term_rewriting::{Context, RuleContext, Signature, parse_rule};
+    /// # fn main() {
+    /// let mut sig = Signature::default();
+    ///
+    /// let mut ops = vec![];
+    /// sig.new_op(2, Some(".".to_string()));
+    /// ops.push(ptp![0, 1; @arrow[tp!(@arrow[tp!(0), tp!(1)]), tp!(0), tp!(1)]]);
+    /// sig.new_op(2, Some("PLUS".to_string()));
+    /// ops.push(ptp![@arrow[tp!(int), tp!(int), tp!(int)]]);
+    /// sig.new_op(1, Some("SUCC".to_string()));
+    /// ops.push(ptp![@arrow[tp!(int), tp!(int)]]);
+    /// sig.new_op(0, Some("ZERO".to_string()));
+    /// ops.push(ptp![int]);
+    ///
+    /// let rules = vec![
+    ///     parse_rule(&mut sig, "PLUS(x_ ZERO) = x_").expect("parsed rule"),
+    ///     parse_rule(&mut sig, "PLUS(x_ SUCC(y_)) = SUCC(PLUS(x_ y_))").expect("parsed rule"),
+    /// ];
+    ///
+    /// let vars = vec![
+    ///     ptp![int],
+    ///     ptp![int],
+    ///     ptp![int],
+    /// ];
+    ///
+    /// println!("{:?}", sig.operators());
+    /// for op in sig.operators() {
+    ///     println!("{:?}/{}", op.name(&sig), op.arity(&sig))
+    /// }
+    /// for r in &rules {
+    ///     println!("{:?}", r);
+    /// }
+    /// let lexicon = Lexicon::from_signature(sig.clone(), ops, vars, vec![], false);
+    ///
+    /// let mut trs = TRS::new(&lexicon, rules).unwrap();
+    ///
+    /// let pretty_before = trs.pretty_utrs(&sig);
+    ///
+    /// let mut rng = thread_rng();
+    ///
+    /// let new_trs = trs.randomly_move_rule(&mut rng).expect("failed when moving rule");
+    ///
+    /// assert_ne!(pretty_before, new_trs.pretty_utrs(&sig));
```

Comment:
Hmm... we probably don't need all three of these tests. Perhaps remove the second one?